lib/rust/ensogl/example/profiling-run-graph/src/lib.rs

```diff
@@ -97,16 +106,64 @@ fn init_theme(scene: &Scene) {
     let theme_manager = theme::Manager::from(&scene.style_sheet);
 
     let theme = theme::Theme::new();
-    theme.set("flame_graph_block_color_active", color::Lcha::blue_green(0.5, 0.8));
-    theme.set("flame_graph_block_color_paused", color::Lcha::blue_green(0.8, 0.0));
-    theme.set("flame_graph_mark_color", color::Lcha::blue_green(0.9, 0.1));
+    theme.set(COLOR_BLOCK_ACTIVE, color::Lcha::blue_green(0.5, 0.8));
+    theme.set(COLOR_BLOCK_PAUSED, color::Lcha::blue_green(0.8, 0.0));
+    theme.set(COLOR_FPS_BAD, color::Lcha::red(0.4, 0.5));
+    theme.set(COLOR_FPS_GOOD, color::Lcha::green(0.8, 0.5));
+    theme.set(COLOR_FPS_MEDIUM, color::Lcha::yellow(0.6, 0.5));
+    theme.set(COLOR_MARK_DEFAULT, color::Lcha::blue_green(0.9, 0.1));
 
     theme_manager.register("theme", theme);
-
     theme_manager.set_enabled(&["theme".to_string()]);
+}
+
+
 
-    let style_watch = ensogl_core::display::shape::StyleWatch::new(&scene.style_sheet);
-    style_watch.get("flame_graph_color");
+// ===========================
+// === Metadata Processing ===
+// ===========================
+
+
+// Create marks for metadata. This will skip `RenderStats` as they are added separately as blocks.
+fn make_marks_from_profile(profile: &Profile<Metadata>) -> Vec<profiler_flame_graph::Mark> {
+    profile
+        .iter_metadata()
+        .filter_map(|metadata: &enso_profiler_data::Metadata<Metadata>| {
+            let position = metadata.mark.into_ms();
+            match metadata.data {
+                Metadata::RenderStats(_) => None,
+                _ => {
+                    let label = metadata.data.to_string();
+                    Some(profiler_flame_graph::Mark { position, label })
+                }
+            }
+        })
+        .collect()
+}
+
+fn make_rendering_performance_blocks(
+    profile: &Profile<Metadata>,
+) -> Vec<profiler_flame_graph::Block> {
+    let mut blocks = Vec::default();
+    let render_stats = profile
+        .iter_metadata()
+        .filter(|metadata| matches!(metadata.data, Metadata::RenderStats(_)));
+    for (prev, current) in render_stats.tuple_windows() {
+        if let Metadata::RenderStats(data) = current.data {
+            let start = prev.mark.into_ms();
+            let end = current.mark.into_ms();
+            let row = -1;
+            let label = format!("{:#?}", data);
+            let theme_color = match data.fps {
+                fps if fps > 55.0 => COLOR_FPS_GOOD,
+                fps if fps > 25.0 => COLOR_FPS_MEDIUM,
```

Comment:
I don't think "medium" is the right word between "good" and "bad". I cannot think of a good word now, but "medium" sounds off too me. To be confirmed with a native speaker.

Reply:
I changed the terminology a bit and instead of “FPS” I am using `Performance`. For that, “Medium” seems to fit better.